### PR TITLE
CLOUDP-92301: Add the implementation of the services that were reused in the atlas go client [ops-manager go client]

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-92301

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by gssbzn — CHANGES_REQUESTED** [review]
don't copy paste the things that are the same

for example, you can reuse the public interface and public structs, you just need to change the implementation

File: opsmngr/accesslist_api_keys.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	if l := root.Links; l != nil {
+		resp.Links = l
+	}
+
+	return root, resp, nil
+}
+
+// Get retrieve information on a single API Key access list entry using the unique identifier for the API Key and desired permitted address.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/api-access-list/get-one-api-access-entry/
```
Comment (gssbzn) [inline opsmngr/accesslist_api_keys.go:69]: wrong link?

File: opsmngr/alert_configurations.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+	"fmt"
+	"net/http"
+
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const alertConfigurationPath = "api/public/v1.0/groups/%s/alertConfigs"
+
+// AlertConfigurationsServiceOp handles communication with the AlertConfiguration related methods
+// of the MongoDB Atlas API.
```
Comment (gssbzn) [inline opsmngr/alert_configurations.go:28]: copy/paste?
```suggestion
// of the MongoDB Ops Manager API.
```

File: opsmngr/continuous_restore_jobs_test.go [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"testing"
+
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+
+	"github.com/go-test/deep"
+)
```
Comment (gssbzn) [inline opsmngr/continuous_restore_jobs_test.go:26]: [nit] let's group these and make sure sorting is right

File: opsmngr/organization_api_keys.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+type APIKeysServiceOp service
+
+var _ atlas.APIKeysService = &APIKeysServiceOp{}
+
+// APIKeysResponse is the response from the APIKeysService.List.
+type APIKeysResponse struct {
+	Links      []*atlas.Link  `json:"links,omitempty"`
+	Results    []atlas.APIKey `json:"results,omitempty"`
+	TotalCount int            `json:"totalCount,omitempty"`
+}
```
Comment (gssbzn) [inline opsmngr/organization_api_keys.go:39]: any reason to redefine this? is it different from Atlas?

**Review by gssbzn — COMMENTED** [review]
Main thing a lot of copy/paste with links and comments pointing to atlas

File: opsmngr/organization_api_keys.go (hunk 4 above)
Comment (andreaangiolillo) [reply opsmngr/organization_api_keys.go:39]: in the atlas go client this is called `aPIKeysResponse` (private)

File: opsmngr/accesslist_api_keys.go (hunk 1 above)
Comment (andreaangiolillo) [reply opsmngr/accesslist_api_keys.go:69]: I don't find the API access list in the ops manager doc :/

Comment (andreaangiolillo) [reply opsmngr/accesslist_api_keys.go:69]: there is only with whitelist https://docs.opsmanager.mongodb.com/current/reference/api/org-api-key-whitelists/

Comment (gssbzn) [reply opsmngr/accesslist_api_keys.go:69]: this is going to be released with 5.0 I was going to suggest the cloud manager links but not there either, may be worth reaching out to docs to confirm when these will be available, ok to leave blank for now

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**